tutorials/download_model_weights.md

```diff
@@ -0,0 +1,230 @@
+garage-bAInd/Platypus2-70B
+garage-bAInd/Camel-Platypus2-13B
+garage-bAInd/Camel-Platypus2-70B
+garage-bAInd/Stable-Platypus2-13B
+garage-bAInd/Platypus2-70B-instruct
+togethercomputer/LLaMA-2-7B-32K
+microsoft/phi-1_5
+microsoft/phi-2
+mistralai/Mistral-7B-v0.1
+mistralai/Mistral-7B-Instruct-v0.1
+mistralai/Mixtral-8x7B-v0.1
+mistralai/Mixtral-8x7B-Instruct-v0.1
+mistralai/Mistral-7B-Instruct-v0.2
+TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
+TinyLlama/TinyLlama-1.1B-Chat-v1.0
+Trelis/Llama-2-7b-chat-hf-function-calling-v2
+```
+
+&nbsp;
+### 2. Download Model Weights
+
+To download the weights for a specific model, use the `--repo_id` argument. Replace `<repo_id>` with the model's repository ID. For example:
+
+```bash
+python litgpt/scripts/download.py --repo_id <repo_id>
+```
+This command downloads the model checkpoint into the `checkpoints/` directory.
+
+&nbsp;
+### 3. Additional Help
+
+
+For more options, add the `--help` flag when running the script:
+
+```bash
+python litgpt/scripts/download.py --help
+```
+
+&nbsp;
+### 4. Run the Model
+
+After conversion, run the model with the `--checkpoint_dir` flag, adjusting `repo_id` accordingly:
+
+```bash
+python chat/base.py --checkpoint_dir checkpoints/repo_id
+```
+
+&nbsp;
+## Tinyllama Example
+
+This section shows a typical end-to-end example for downloading and using TinyLlama:
+
+1. List available TinyLlama checkpoints:
+
+```bash
+python litgpt/scripts/download.py | grep Tiny
+```
+
+```
+TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
+TinyLlama/TinyLlama-1.1B-Chat-v1.0
+```
+
+2. Download a TinyLlama checkpoint:
+
+```bash
+export repo_id=TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T
+python litgpt/scripts/download.py --repo_id $repo_id
+```
+
+3. Use the TinyLlama model:
+
+```bash
+python chat/base.py --checkpoint_dir checkpoints/$repo_id
+```
+
+&nbsp;
+## Specific Models
+
+Note that certain models require that you've been granted access to the weights on the HuggingFace Hub. 
+
+For example, to get access to the Gemma 2B model, you can do so by following the steps at https://huggingface.co/google/gemma-2b. After access is granted, you can find your HF hub token in https://huggingface.co/settings/tokens.
+
+Once you've been granted access and obtained the access token you need to pass the additional `--access_token`. Some models like Gemma also only support the safetensors format, which you can specify via the `--from_safetensors true` flag:
+
+```bash
+python litgpt/scripts/download.py \
+  --repo_id google/gemma-2b \
+  --access_token your_hf_token \
+  --from_safetensors true
+```
+
+
+&nbsp;
+## Tips for GPU Memory Limitations
+
+The `download.py` script will automatically convert the downloaded model checkpoint into a LitGPT-compatible format. In case this conversion fails due to GPU memory constraints, you can reduce the memory requirements by passing the  `--dtype bfloat16` flag to convert all parameters into this smaller precision:
```

Comment:
I must say that I think almost all of the weight we support come in 16 bit already

Reply:
Sure, but it doesn't hurt to add a note. Thanks. Will update.